src/publish/index.coffee

```diff
@@ -47,48 +47,51 @@ module.exports = (dir, log, config, version, testCommand) ->
     verify dir, (error) ->
       endIf(error)
 
-      test dir, log, npm, testCommand, (error) ->
+      npm.install config.registry, (error) ->
```

Comment:
It gets really hard to see what actually changed in here. At some point an `async.series` or similar might help make it easier to follow/track.